tests/matrix_free/laplace_operator_01.cc

```diff
@@ -108,7 +108,7 @@ void test ()
   }
 
   MatrixFreeOperators::LaplaceOperator<dim,fe_degree,fe_degree+1, 1, number> mf;
-  mf.initialize(mf_data);
+  mf.initialize(std_cxx11::make_shared<MatrixFree<dim,number> >(mf_data));
```

Comment:
Here (and in a number of tests below) you create a `MatrixFree` objects (in line 100) and later call `make_shared` on that. Would you mind creating it as a `shared_ptr` straight away? One of the motivations of the `shared_ptr` was to be able to forget about the `MatrixFree` object in the calling class. In other words, I would like to write code as follows for this test:
```
MatrixFreeOperators::LaplaceOperator<dim,fe_degree,fe_degree+1, 1, number> mf;
{
  std_cxx11::shared_ptr<MatrixFree<dim,number> > mf_data (new MatrixFree<dim,number>());
  const QGauss<1> quad (fe_degree+1);
  typename MatrixFree<dim,number>::AdditionalData data;
  data.tasks_parallel_scheme =
    MatrixFree<dim,number>::AdditionalData::none;
  data.tasks_block_size = 7;
  mf_data->reinit (dof, constraints, quad, data);
  mf.initialize(mf_data);
}
```
Would you mind changing that? I can also do that in my follow-up PR that cleans up `step-37`.